src/main/java/com/minecolonies/coremod/colony/requestsystem/data/StandardRequestableTypeRequestResolverAssignmentDataStore.java

```diff
@@ -0,0 +1,127 @@
+package com.minecolonies.coremod.colony.requestsystem.data;
+
+import com.google.common.reflect.TypeToken;
+import com.minecolonies.api.colony.requestsystem.StandardFactoryController;
+import com.minecolonies.api.colony.requestsystem.data.IRequestableTypeRequestResolverAssignmentDataStore;
+import com.minecolonies.api.colony.requestsystem.factory.FactoryVoidInput;
+import com.minecolonies.api.colony.requestsystem.factory.IFactory;
+import com.minecolonies.api.colony.requestsystem.factory.IFactoryController;
+import com.minecolonies.api.colony.requestsystem.token.IToken;
+import com.minecolonies.api.util.NBTUtils;
+import com.minecolonies.api.util.constant.NbtTagConstants;
+import com.minecolonies.api.util.constant.TypeConstants;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.Tuple;
+import net.minecraftforge.common.util.Constants;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class StandardRequestableTypeRequestResolverAssignmentDataStore implements IRequestableTypeRequestResolverAssignmentDataStore
+{
+
+    private IToken<?> id;
+    private final Map<TypeToken<?>, Collection<IToken<?>>> assignments;
+
+    public StandardRequestableTypeRequestResolverAssignmentDataStore(
+      final IToken<?> id,
+      final Map<TypeToken<?>, Collection<IToken<?>>> assignments) {
+        this.id = id;
+        this.assignments = assignments;
+    }
+
+    public StandardRequestableTypeRequestResolverAssignmentDataStore()
+    {
+        this(StandardFactoryController.getInstance().getNewInstance(TypeConstants.ITOKEN), new HashMap<>());
+    }
+
+    @NotNull
+    @Override
+    public Map<TypeToken<?>, Collection<IToken<?>>> getAssignments()
+    {
+        return assignments;
+    }
+
+    @Override
+    public IToken<?> getId()
+    {
+        return id;
+    }
+
+    @Override
+    public void setId(final IToken<?> id)
+    {
+        this.id = id;
+    }
+
+    public static class Factory implements IFactory<FactoryVoidInput, StandardRequestableTypeRequestResolverAssignmentDataStore>
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 56 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)